src/wrapper.rs

```diff
@@ -0,0 +1,102 @@
+/* SPDX-License-Identifier: MIT */
+/*
+ * Copyright (C) 2023 IBM
+ *
+ * Authors:
+ *   Claudio Carvalho <cclaudio@ibm.com>
+ *   Vikram Narayanan <vikram186@gmail.com>
+ */
+
+#![allow(non_camel_case_types)]
+
+use crate::bindings::strlen;
+use crate::mem::sizeof_alloc;
+use crate::prints;
+use crate::util::util::memset;
+use alloc::alloc::{alloc, dealloc};
+use core::alloc::Layout;
+use core::slice;
+use core::{ptr, str};
+
+pub type c_char = u8;
+
+const ALIGN_32: usize = 32;
+
+#[no_mangle]
+pub extern "C" fn malloc(sz: usize) -> *mut u8 {
+    let layout = match Layout::from_size_align(sz, ALIGN_32) {
+        Ok(l) => l,
+        Err(_e) => panic!("malloc: size is not 32 bytes aligned"),
+    };
+
+    let ptr: *mut u8 = unsafe { alloc(layout) };
+    ptr
+}
+
+#[no_mangle]
+pub extern "C" fn calloc(items: usize, sz: usize) -> *mut u8 {
+    let num_bytes: usize = items * sz;
```

Comment:
Before computing num_bytes check if items * sz will overflow. I think you can check `items.checked_mul(sz).is_none()` and if so return an error